src/asm/section.cpp

```diff
@@ -307,6 +308,8 @@ static struct Section *getSection(char const *name, enum SectionType type, uint3
 	uint8_t alignment = attrs->alignment;
 	uint16_t alignOffset = attrs->alignOfs;
 
+	assert(alignment < 16 && alignOffset < 1 << alignment); // Should be ensured by the caller
```

Comment:
`getSection` is only called by `sect_NewSection` and `sect_SetLoadSection`, and both of those are only called in parser.y, after parsing a `sectattrs`, which may have parsed an `align_spec`, which does its own bounds checking and error-messaging. (A couple other functions in section.cpp already have this assertion.)